src/lib.js

```diff
@@ -677,6 +676,65 @@ var program,
       }
     },
 
+    /**
+     * Identifies whether the input data string is a graphql query or not
+     *
+     * @param {string} dataString - Input data string to check if it is a graphql query
+     * @param {string} contentType - Content type header value
+     * @returns {Object} - { result: true, graphql: {Object} } if dataString is a graphql query else { result: false }
+    */
+    identifyGraphqlRequest: function (dataString, contentType) {
+      try {
+        const rawDataObj = _.attempt(JSON.parse, dataString.replace(/[\r\n]/g, ''));
+        if (contentType === 'application/json' && rawDataObj && !_.isError(rawDataObj)) {
+          if (!_.has(rawDataObj, 'query') || !_.isString(rawDataObj.query)) {
+            return { result: false };
+          }
+          if (!(/^[^]*\{[^]*\}$/g).test(rawDataObj.query)) {
```

Comment:
@aman-v-singh What is this regexp supposed to do? Do we need this? As discussed if we can solve this without regexp it'd be better solution as we'll be checking this almost all of times due to body being most used format  `application/json`.

Reply:
@VShingala without regex it would be difficult to distinguish between a normal string and a `graphql` query.
Data like `--data '{"query": "test"}' ` will also be opening in the `graphql` tab.

Reply:
@aman-v-singh Which is completely fine as we discussed. i.e. We can have this edge case handled in future if we come across it but it is very uncommon to have this scenario. Main objective is to not slow down performance for most of imports to handle few edge case.